rocket-mission-persistence/src/main/java/cz/muni/fi/dao/MissionDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package cz.muni.fi.dao;
+
+import cz.muni.fi.entity.Mission;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.List;
+
+/**
+ * Data Access Object implementation for Mission entity.
+ *
+ * @author Martin Kazimir
+ */
+@Repository
+@Transactional
+public class MissionDaoImpl implements MissionDao {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
```

Comment:
Nechceme v repository obecně ty checky na null v metodách? Možná by na to šla vytvořit bokem Guard class, ať nemáme zbytečný duplicitní kód.